pkg/controller/dnsendpoint/ingress.go

```diff
@@ -0,0 +1,99 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package dnsendpoint
+
+import (
+	"fmt"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	pkgruntime "k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/watch"
+	restclient "k8s.io/client-go/rest"
+
+	feddnsv1a1 "github.com/kubernetes-sigs/federation-v2/pkg/apis/multiclusterdns/v1alpha1"
+	fedclientset "github.com/kubernetes-sigs/federation-v2/pkg/client/clientset/versioned"
+)
+
+func StartIngressDNSEndpointController(config *restclient.Config, stopChan <-chan struct{}, minimizeLatency bool) error {
+	restclient.AddUserAgent(config, "Ingress DNSEndpoint")
+	client := fedclientset.NewForConfigOrDie(config)
+
+	listFunc := func(options metav1.ListOptions) (pkgruntime.Object, error) {
+		return client.MulticlusterdnsV1alpha1().MultiClusterIngressDNSRecords(metav1.NamespaceAll).List(options)
+	}
+	watchFunc := func(options metav1.ListOptions) (watch.Interface, error) {
+		return client.MulticlusterdnsV1alpha1().MultiClusterIngressDNSRecords(metav1.NamespaceAll).Watch(options)
+	}
+
+	controller, err := newDNSEndpointController(client, &feddnsv1a1.MultiClusterIngressDNSRecord{}, "ingress",
+		listFunc, watchFunc, getIngressDNSEndpoints, minimizeLatency)
+	if err != nil {
+		return err
+	}
+
+	go controller.Run(stopChan)
+	return nil
+}
+
+// getIngressDNSEndpoints returns endpoint objects for each MultiClusterIngressDNSRecord object that should be processed.
+func getIngressDNSEndpoints(obj interface{}) ([]*feddnsv1a1.Endpoint, error) {
+	var endpoints []*feddnsv1a1.Endpoint
+
+	dnsObject, ok := obj.(*feddnsv1a1.MultiClusterIngressDNSRecord)
+	if !ok {
+		return nil, fmt.Errorf("receieved event for unknown object %v", obj)
```

Comment:
s/receieved/received/

Reply:
oops, i keep doing this mistake :). now fixed.